### PR TITLE
phpstan: fixes for Varien_Data_Collection

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

phpdocs/phpstan fix.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Adminhtml/Block/Report/Grid.php [hunk 1]
```diff
@@ -21,6 +21,8 @@
 /**
  * Adminhtml report grid block
  *
+ * @method Mage_Reports_Model_Resource_Report_Collection getCollection()
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/Report/Grid.php:24]: Can we put these lines below the 3 lines (category package and author)? so the "header" stays always "description, empty line, category, package, author" and then everything else?

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Report/Grid.php:24]: Yes we can, but there are a files ... 

(when i added methods i did not really care about this :()

Comment (fballiano) [reply app/code/core/Mage/Adminhtml/Block/Report/Grid.php:24]: I'm fixing all of them with the PRs that I submitted today (and I'll continue tonight and tomorrow) ;-)

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Report/Grid.php:24]: When i added new @methods i tried to sort them in alphabetically order ... :P

Comment (fballiano) [reply app/code/core/Mage/Adminhtml/Block/Report/Grid.php:24]: that's fine, I never changed that order, I just move them after the "header/copyright" part :-)

**Review by fballiano — APPROVED** [review]

**Comment by kiatng** [comment]
Are you going to fix this:

image

If the filtering is enabled in the grid, then the collection should have  `Varien_Data_Collection_Db` as parent:

```php
     * @param Varien_Data_Collection|Varien_Data_Collection_Db $collection
     */
    public function getCollection($collection)
    // and: public function setCollection($collection)
```

**Comment by sreichel** [comment]
> Do you have plan to fix this:

Yes. The "problem" is, that is has no impact on phpstan ... so i dont see it during my tests. Made a new PR. :)

**Review by kiatng — APPROVED** [review]

**Pull request merged**